packages/application-extension/src/index.tsx

```diff
@@ -63,22 +63,24 @@ namespace Patterns {
  */
 const main: JupyterLabPlugin<void> = {
   id: '@jupyterlab/application-extension:main',
-  requires: [ICommandPalette, IRouter],
-  activate: (app: JupyterLab, palette: ICommandPalette, router: IRouter) => {
+  requires: [ICommandPalette, IRouter, IWindowResolver],
+  activate: (app: JupyterLab, palette: ICommandPalette, router: IRouter, resolver: IWindowResolver) => {
+    // Requiring the window resolver guarantees that the application extension
+    // only loads if there is a viable window name. Otherwise, the application
+    // will short-circuit and ask the user to navigate away.
+    const workspace = resolver.name ? `"${resolver.name}"` : '[default: /lab]';
+
+    console.log(`Starting ${main.id} in workspace ${workspace}`);
```

Comment:
Do you intend this `console.log`?

Reply:
Yeah, I wanted a concrete use of the `IWindowResolver` so that it isn't just being included for its side-effects. I'm not sure about this, what's your opinion?

Reply:
I think it's reasonable, but I view the plugin `id`s as being more developer-facing implementation details that should not necessarily be displayed to the user (as opposed to workspace names). Maybe just
```javascript
console.log(`Starting workspace ${workspace}`)?
```